doc/source/run_engine_api.rst

```diff
@@ -0,0 +1,56 @@
+.. currentmodule:: bluesky.run_engine
+
+RunEngine API Documentation
+===========================
+
+.. autoclass:: RunEngine
+
+    .. automethod:: __call__
+
+    The RunEngine maintains a callback registry of functions that receive any
+    :doc:`documents` generated by plan execution. These methods add and remove
+    functions from that registry.
+
+    .. automethod:: subscribe
```

Comment:
I think using `autosummary` here would be better so each method gets it's own page and these are just tables. 

Not a blocking concern.